src/main/java/com/arpnetworking/metrics/mad/parsers/IsmClientProtocolRecordParser.java

```diff
@@ -0,0 +1,129 @@
+        try {
+            final ClientV1.RecordSet request = ClientV1.RecordSet.parseFrom(data.getBody());
+            final List<Record> records = Lists.newArrayList();
+            for (ClientV1.Record record : request.getRecordsList()) {
+                final ByteBuffer byteBuffer = ByteBuffer.wrap(record.getId().toByteArray());
+                final Long high = byteBuffer.getLong();
+                final Long low = byteBuffer.getLong();
+                final DefaultRecord.Builder builder = new DefaultRecord.Builder()
+                        .setId(new UUID(high, low).toString())
+                        .setTime(new DateTime(record.getEndMillisSinceEpoch()))
+                        .setAnnotations(buildAnnotations(record))
+                        .setMetrics(buildMetrics(record));
+
+                records.add(builder.build());
+            }
+            return records;
+        } catch (final InvalidProtocolBufferException e) {
+            throw new ParsingException("Could not create Request message from data", data.getBody(), e);
+        } catch (final ConstraintsViolatedException | IllegalArgumentException e) {
+            throw new ParsingException("Could not build record", data.getBody(), e);
+        }
+    }
+
+    private ImmutableMap<String, ? extends Metric> buildMetrics(final ClientV1.Record record) {
+        final ImmutableMap.Builder<String, Metric> metrics = ImmutableMap.builder();
+        processEntries(metrics, record.getCountersList(), MetricType.COUNTER);
+        processEntries(metrics, record.getTimersList(), MetricType.TIMER);
+        processEntries(metrics, record.getGaugesList(), MetricType.GAUGE);
+        return metrics.build();
+    }
+
+    private void processEntries(
+            final ImmutableMap.Builder<String, Metric> metrics,
+            final List<ClientV1.MetricEntry> entries,
+            final MetricType metricType) {
+        for (ClientV1.MetricEntry metricEntry : entries) {
+            final DefaultMetric.Builder metricBuilder = new DefaultMetric.Builder()
+                    .setType(metricType);
+            final List<Quantity> quantities = Lists.newArrayList();
+            for (ClientV1.DoubleQuantity quantity : metricEntry.getSamplesList()) {
+                quantities.add(
+                        new Quantity.Builder()
+                                .setUnit(baseUnit(quantity.getUnit()))
+                                .setValue(quantity.getValue())
+                                .build());
+            }
+
+            metricBuilder.setValues(quantities);
+            metrics.put(metricEntry.getName(), metricBuilder.build());
+        }
+    }
+
+    @Nullable
+    private Unit baseUnit(final ClientV1.CompoundUnit compoundUnit) {
+        if (compoundUnit.getNumeratorCount() > 0) {
+            final ClientV1.Unit selectedUnit = compoundUnit.getNumerator(0);
+            if (selectedUnit.getType() == ClientV1.Unit.Type.UNRECOGNIZED) {
```

Comment:
I would prefer .equsls, with the constant on the left side. == always assumes something about the implementation (e.g. enum vs static instance).